packages/common/src/services/audius-backend/waudio.ts

```diff
@@ -1,94 +1,102 @@
-import { AccountInfo } from '@solana/spl-token'
-import { PublicKey } from '@solana/web3.js'
+import { AudiusLibs } from '@audius/sdk'
 
-import { Nullable } from 'utils/typeUtils'
-
-import { AnalyticsEvent, Name } from '../../models'
+import { AnalyticsEvent, Name, SolanaWalletAddress } from '../../models'
 
 import { AudiusBackend } from './AudiusBackend'
 
-export const deriveUserBank = async (audiusBackendInstance: AudiusBackend) => {
-  const audiusLibs = await audiusBackendInstance.getAudiusLibs()
-  return (await audiusLibs.solanaWeb3Manager.deriveUserBank()) as PublicKey
-}
-
-export const doesUserBankExist = async (
-  audiusBackendInstance: AudiusBackend
-) => {
-  const audiusLibs = await audiusBackendInstance.getAudiusLibs()
-  const userBank: PublicKey =
-    await audiusLibs.solanaWeb3Manager.deriveUserBank()
-  const doesExist = await checkIsCreatedTokenAccount(
-    userBank.toString(),
-    audiusBackendInstance
-  )
-  return doesExist
-}
-
-export const checkIsCreatedTokenAccount = async (
-  addr: string,
-  audiusBackendInstance: AudiusBackend
+export const deriveUserBankPubkey = async (
+  audiusBackendInstance: AudiusBackend,
+  sourceEthAddress?: string
 ) => {
-  const audiusLibs = await audiusBackendInstance.getAudiusLibs()
-  const tokenAccount: Nullable<AccountInfo> =
-    await audiusLibs.solanaWeb3Manager.getTokenAccountInfo(addr)
-  return tokenAccount != null
+  const audiusLibs: AudiusLibs = await audiusBackendInstance.getAudiusLibs()
+  return await audiusLibs.solanaWeb3Manager!.deriveUserBank(sourceEthAddress)
 }
 
-export const createUserBank = async (
-  feePayerOverride = null,
-  audiusBackendInstance: AudiusBackend
+export const deriveUserBankAddress = async (
```

Comment:
Made this into two methods, one that returns the pubkey and one that returns a branded `SolanaWalletAddress` for extra safety